File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n,vector<int>& dp){
        if(n==1){
            return 1;
        }
        if(n==2){
            return 2;
        }
        if(dp[n] != -1){
            return dp[n];
        }
        return dp[n] = climbStairs(n-1,dp)+climbStairs(n-2,dp);
    }
    int climbStairs(int n) {
        vector<int> dp(n+1,-1);
        return climbStairs(n,dp);
        return dp[n];
    }
};

File: 0070-climbing-stairs/0070-climbing-stairs.js
/**
 * @param {number} n
 * @return {number}
 */

function helper(n, dp){
    if(n < 0){
        return 0;
    }
    if(n == 0){
        return 1;
    }
    if(dp[n] !== -1){
        return dp[n];
    }
    dp[n] = helper(n-1, dp)+helper(n-2, dp);
    return dp[n];
}
var climbStairs = function(n) {
    const dp = new Array(n+1).fill(-1);
    return helper(n, dp);
};